components/lib/datatable/datatable.d.ts

```diff
@@ -212,33 +212,56 @@ interface DataTableRowExpansionTemplate {
     customRendering: boolean;
 }
 
-interface DataTableRowClassNameOptions {
-    props: DataTableProps;
+interface DataTableRowClassNameOptions<TValue extends DataTableValueArray> {
+    props: DataTableProps<TValue>;
 }
 
-interface DataTableCellClassNameOptions {
-    props: DataTableProps;
+interface DataTableCellClassNameOptions<TValue extends DataTableValueArray> {
+    props: DataTableProps<TValue>;
     rowData: any;
```

Comment:
Did you miss an any here?

Reply:
Yes I did. I'll push the change but should I also rename `DataTableRowSelection` to something else since its used here too?

Reply:
I have to look but there isn't it just a Value Array not a Selection Array?

Reply:
Looks like it gets passed to `ObjectUtils.getPropValue` in `BodyCell` but I am having a hard time figuring out what it's doing.

Reply:
```
options:{
column: Object { "$$typeof": Symbol("react.element"), type: Column()
, key: ".$=2R2sm=2code", … }
field: "code"
frozenRow: false
props: Object { dataKey: "id", selectionMode: "single", cellSelection: true, … }
rowIndex: 0
}
```

It looks like rowData doesn't exist in options. This is what I get as an output. Not sure if there is a specific usecase where it will show up but it seems like the rest of the fields in options are also not typed. Let me know how to proceed with this.

Reply:
You are right i think that definiton is wrong I think it should be this right?

```
interface DataTableCellClassNameOptions {
    props: DataTableProps;
    column: Column;
    field: string;
    frozenRow: boolean;
    rowIndex: number;
}
```
?

Reply:
Looks correct to me